File: api_server/models/oauth.py
from . import database, db


class Oauth(database.Model):
    user_id = db.Column(db.Integer, db.ForeignKey('user.user_id'), primary_key=True)
    oauth_adapter = db.Column(db.Text, primary_key=True)
    embedding_user = db.Column(db.Text)
    user = db.relationship(
        'User',
        uselist=False
    )

File: RoseliaCore/Database/Models/OAuth.fs
namespace RoseliaBlog.RoseliaCore.Database.Models

open System.ComponentModel.DataAnnotations
open System.ComponentModel.DataAnnotations.Schema

[<CLIMutable>]
[<Table("oauth")>]
type OAuth = {
    [<Column("user_id")>]
    UserId: int
    
    [<Column("oauth_adapter")>]
    OAuthAdapter: string
    
    [<Column("embedding_user")>]
    UnderlyingUser: string
    
    [<ForeignKey("UserId")>]
    User: User
}
